### ISSUE TITLE
Can not reproduce the results of SWE-Gym

### ISSUE THREAD

**Issue by junleiz** [SWE-Gym/SWE-Gym#20]
Thank you for your work. I tried to use the swe-gym dataset, which can be found at [https://huggingface.co/datasets/SWE-Gym/OpenHands-SFT-Trajectories/viewer](https://huggingface.co/datasets/SWE-Gym/OpenHands-SFT-Trajectories/viewer), to fine-tune Qwen3-Coder-flash. I then tested the model's performance on 50 sampled examples from SWE-bench-Verified, but the results were very poor. This is my `verl` code:

```
#!/bin/bash
set -x

if [ "$#" -lt 1 ]; then
    echo "Usage: run_swe_gym_sft.sh <nproc_per_node> [other_configs...]"
    echo "Example: bash examples/sft/swe_gym/run_swe_gym_sft.sh 8"
    exit 1
fi

nproc_per_node=$1
model_path=/disk2/models/qwen/Qwen2.5-Coder-32B-Instruct
save_path=/WillDevExt/models/qwen/swe_gym_sft_Qwen2.5-Coder-32B-Instruct_run2
TRAIN_FILE=/home/yueyang/willfarm/data/swe_gym_parquet/swe_gym_sft_train.parquet
VAL_FILE=/home/yueyang/willfarm/data/swe_gym_parquet/swe_gym_sft_val_split.parquet
# Shift the arguments so $@ refers to the rest
shift 1


# Run SFT training for SWE Gym dataset
# Note: For 32B model on 8 A100 GPUs, we use FSDP2 instead of tensor parallelism
# FSDP2 provides better memory efficiency for SFT training by sharding model parameters
# Tensor parallelism is mainly used in PPO/GRPO trainers with Megatron backend
torchrun --nnodes=1 --nproc_per_node=$nproc_per_node --master_addr localhost --master_port 12743 \
 -m verl.trainer.fsdp_sft_trainer \
    data.train_files=$TRAIN_FILE \
    data.val_files=$VAL_FILE \
    data.train_batch_size=8 \
    data.micro_batch_size_per_gpu=1 \
    data.multiturn.enable=true \
    data.multiturn.messages_key=messages \
    data.max_length=32768 \
    data.truncation=left \
    model.enable_gradient_checkpointing=true \
    model.strategy=fsdp2 \
    model.fsdp_config.model_dtype=bf16 \
    model.fsdp_config.cpu_offload=false \
    model.partial_pretrain=$model_path \
    optim.lr=1e-5 \
    optim.lr_scheduler=cosine \
    optim.warmup_steps_ratio=0.02 \
    trainer.default_local_dir=$save_path \
    trainer.project_name=swe-gym-sft \
    trainer.experiment_name=swe-gym-sft-qwen2.5-coder-32b-instruct \
    trainer.logger=['console','wandb'] \
    trainer.total_epochs=5 \
    trainer.save_freq=100 \
    trainer.test_freq=100 \
    trainer.logger=['console','wandb'] \
    trainer.default_hdfs_dir=null \
    ulysses_sequence_parallel_size=4 \
    use_remove_padding=true  $@
```

I compared my trained model with https://huggingface.co/SWE-Gym/OpenHands-32B-Agent and found that, on our sampled SWE-bench dataset of size 50, the former can solve 11 cases, while my trained model can only solve 6. I tried tuning multiple sets of hyperparameters, but none of them could exceed 6.

**Comment by Jiayi-Pan** [comment]
hi!

1. openhands framework changes frequently, our data/evals are done in this version https://github.com/SWE-Gym/OpenHands
2. we are not sure how well qwen-3 works / how to best tune its hyper-parameter. please use qwen 2.5 coder, as disucssued in the paper, for reproduction.

**Comment by junleiz** [comment]
I am sorry for the mistake. Actually i tune the Qwen2.5 Coder as shown in the scripts. I used the openhands-32B-Agent running on openhands v0.44 as the baseline. And try to tune the Qwen2.5 Coder Instruct to achieve the comparable results. But we failed to achieve the results. Do you think it is because the version of Openhands? Thanks!

---- Replied Message ----
| From | Jiayi ***@***.***> |
| Date | 08/12/2025 01:05 |
| To | SWE-Gym/SWE-Gym ***@***.***> |
| Cc | Junlei ***@***.***>,
Author ***@***.***> |
| Subject | Re: [SWE-Gym/SWE-Gym] Can not reproduce the results of SWE-Gym (Issue #20) |
Jiayi-Pan left a comment (SWE-Gym/SWE-Gym#20)

hi!

openhands framework changes frequently, our data/evals are done in this version https://github.com/SWE-Gym/OpenHands
we are not sure how well qwen-3 works / how to best tune its hyper-parameter. please use qwen 2.5 coder, as disucssued in the paper, for reproduction.

**Comment by Jiayi-Pan** [comment]
interesting. my best guess is verl's sft trainer has some bug (maybe in chat template or sth) causing the mismatch -- could you try torchtune? https://github.com/SWE-Gym/SWE-Gym/tree/main/scripts/training/openhands

**Comment by junleiz** [comment]
Very thanks for your suggestions. I will have a try. The system prompt used during inference (OH v0.44) is totally different from the one used in training. But I re-run your Openhands-32B-Agent as a new baseline. They both inference under the mismatch system prompt setting. Do you think it is a problem? Thanks!

Furthermore, actually, I can achieve a slightly lower result under the setting of first achieve (temperature=0, iterative eval mode=false) compared with Openhands-32B-Agent, but failed in the iterative mode (3 attempts) setting.
